frame/staking/src/lib.rs

```diff
@@ -2307,19 +2314,29 @@ pub mod pallet {
 			let ledger = Self::ledger(&controller).ok_or(Error::<T>::NotController)?;
 			let stash = ledger.stash;
 
-			// If the caller is not the controller, we want to check that the minimum bond
-			// requirements are not satisfied, and thus we have reason to chill this user.
+			// If the caller is not the controller, we first check that we are near our threshold
+			// limit, and then we also check that the minimum bond requirements are not satisfied,
+			// and thus we are justified to chill this user.
 			//
 			// Otherwise, if caller is the same as the controller, this is just like `chill`.
 			if caller != controller {
 				let min_active_bond = if Nominators::<T>::contains_key(&stash) {
+					let max_nominator_count = MaxNominatorsCount::<T>::get().ok_or(Error::<T>::CannotChillOther)?;
+					let current_nominator_count = CounterForNominators::<T>::get();
+					// Safe for up to hundreds of millions of users.
```

Comment:
If we are concerned about saturating the multiplication we run into issues at tens of millions:

`2^32 -1 == 4_294_967_295`
`4_294_967_295 / 100 == 42_949_672`

So we would would saturate at `42_949_672` `current_nominator _count` - which seems totally fine, but just wanted to make sure I am understand what the comment is getting at